### PR TITLE
feat(eth-multisig-v2): add support for Avax

### PR DESCRIPTION
**Pull request by CapnMigraine**
CLOSES TICKET: STLX-5121

### CODE REVIEW

File: contracts/coins/AvaxWalletSimple.sol [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+ *
+ * Data Formats
+ * ============
+ *
+ * The signature is created with ethereumjs-util.ecsign(operationHash).
+ * Like the eth_sign RPC call, it packs the values as a 65-byte array of [r, s, v].
+ * Unlike eth_sign, the message is not prefixed.
+ *
+ * The operationHash the result of keccak256(prefix, toAddress, value, data, expireTime).
+ * For ether transactions, `prefix` is "ETHER".
```
Comment (wesgraham) [inline contracts/coins/AvaxWalletSimple.sol:25]: Nit: is this ether?

**Review by wesgraham — APPROVED** [review]

File: contracts/coins/AvaxWalletSimple.sol [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+   * 2 signers will be required to send a transaction from this wallet.
+   * Note: The sender is NOT automatically added to the list of signers.
+   * Signers CANNOT be changed once they are set
+   *
+   * @param allowedSigners An array of signers on the wallet
+   */
+  function AvaxWalletSimple(address[] allowedSigners) public {
+    if (allowedSigners.length != 3) {
+      // Invalid number of signers
+      revert();
```
Comment (ryan-bitgo) [inline contracts/coins/AvaxWalletSimple.sol:62]: i think `require` with a message is better here.

File: contracts/coins/AvaxWalletSimple.sol [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Modifier that will execute internal code block only if the sender is an authorized signer on this wallet
+   */
+  modifier onlySigner {
+    if (!isSigner(msg.sender)) {
```
Comment (ryan-bitgo) [inline contracts/coins/AvaxWalletSimple.sol:86]: same. use `require` instead of `revert`

File: contracts/coins/AvaxWalletSimple.sol [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    _;
+  }
+
+  /**
+   * Gets called when a transaction is received without calling a method
+   */
+  function() public payable {
+    if (msg.value > 0) {
+      // Fire deposited event if we are receiving funds
+      Deposited(msg.sender, msg.value, msg.data);
```
Comment (ryan-bitgo) [inline contracts/coins/AvaxWalletSimple.sol:98]: i think this needs to be prefixed with `emit`.

File: contracts/coins/AvaxWalletSimple.sol [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+    var operationHash = keccak256("AVAX", toAddress, value, data, expireTime, sequenceId);
+    
+    var otherSigner = verifyMultiSig(toAddress, operationHash, signature, expireTime, sequenceId);
+
+    // Success, send the transaction
+    if (!(toAddress.call.value(value)(data))) {
+      // Failed executing transaction
+      revert();
+    }
+    Transacted(msg.sender, otherSigner, operationHash, toAddress, value, data);
```
Comment (ryan-bitgo) [inline contracts/coins/AvaxWalletSimple.sol:139]: same prefix with `emit`

File: contracts/coins/AvaxWalletSimple.sol [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+      uint sequenceId,
+      bytes signature
+  ) public onlySigner {
+    // Verify the other signer
+    var operationHash = keccak256("AVAX", toAddress, value, data, expireTime, sequenceId);
+    
+    var otherSigner = verifyMultiSig(toAddress, operationHash, signature, expireTime, sequenceId);
+
+    // Success, send the transaction
+    if (!(toAddress.call.value(value)(data))) {
```
Comment (ryan-bitgo) [inline contracts/coins/AvaxWalletSimple.sol:135]: same use `require`

**Review by ryan-bitgo — CHANGES_REQUESTED** [review]

File: contracts/coins/AvaxWalletSimple.sol (hunk 3 above)
Comment (CapnMigraine) [reply contracts/coins/AvaxWalletSimple.sol:86]: `require` is not supported in Solidity `0.4.18`
We want to deviate the least possible from the original v2 contract implemented in [WalletSimple.sol](https://github.com/BitGo/eth-multisig-v2/blob/master/contracts/WalletSimple.sol), that way we can get a faster security audit.

File: contracts/coins/AvaxWalletSimple.sol (hunk 1 above)
Comment (CapnMigraine) [reply contracts/coins/AvaxWalletSimple.sol]: fixed

**Review by karankurbur — APPROVED** [review]

**Review by ryan-bitgo — APPROVED** [review]

**Pull request merged**